build/sh_syntax_check.sh

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file ensures that all shell script in the repo have no syntax errors.
+
+#!/usr/bin/env bash
+bad=0
+for script in $(find -name "*.sh"); do
+  bash -n "${script}"
+  if [[ $? != 0 ]]; then
+    bad=1
```

Comment:
Can we move the logic from the next if-statement to here, eliminate the need of using the local variable `bad`, and make the code a bit shorter?

```
  if [[ $? != 0 ]]; then
    echo "Scripts with invalid syntax found, please look at the errors above and fix."
    exit 1
  fi
```

Reply:
No, the idea here is to show all the bad scripts in one go. Otherwise if for some reason multiple scripts are bad, it'll take multiple rounds of  CI/CD runs, commits and lost time. This way you get one lists of bad scripts and can fix them all in one go.

Reply:
gotcha